recipes/libsodium/1.0.18/conanfile.py

```diff
@@ -0,0 +1,223 @@
+    topics = ("sodium", "libsodium", "encryption", "signature", "hashing")
+    generators  = "cmake"
+    _source_subfolder = "source_subfolder"
+
+    options = {
+        "shared" : [True, False],
+        "fPIC": [True, False],
+        "use_soname" : [True, False],
+        "use_pie"    : [True, False],
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_soname": True,
+        "use_pie": False,
+    }
+
+    @property
+    def _android_id_str(self):
+        return "androideabi" if str(self.settings.arch) in ["armv6", "armv7"] else "android"
+
+    @property
+    def _arch_id_str_compiler(self):
+        return {"x86": "i686",
+                "armv6": "arm",
+                "armv7": "arm",
+                "armv7hf": "arm",
+                "armv8": "aarch64",
+                "mips64": "mips64"}.get(str(self.settings.arch),
+                                        str(self.settings.arch))
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _vs_platform(self):
+        if self.settings.arch.value == "x86":
+            return "Win32"
+        return "x64"
+
+    @property
+    def _vs_configuration(self):
+        configuration = ""
+        if self.options.shared:
+            configuration += "Dyn"
+        else:
+            configuration += "Static"
+        build_type = str(self.settings.build_type)
+        if build_type == "RelWithDebInfo":
+            build_type = "Release"
+        configuration += build_type
+        return configuration
+
+    @property
+    def _vs_sln_folder(self):
+        if self.settings.compiler.version == "14":
+            return "vs2015"
+        elif self.settings.compiler.version == "15":
+            return "vs2017"
+        elif self.settings.compiler.version == "16":
+            return "vs2019"
+        else:
+            raise ConanInvalidConfiguration("Unsupported msvc version: {}".format(self.settings.compiler.version))
+
+    @property
+    def _runtime_prefix(self):
+        return str(self.settings.compiler.runtime)[:2]
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _build_visual(self):
+        sln_path = os.path.join(self.build_folder, self._source_subfolder, "builds", "msvc", self._vs_sln_folder, "libsodium.sln")
+
+        msbuild = MSBuild(self)
+        msbuild.build(sln_path, upgrade_project=False, platforms={"x86": "Win32"}, build_type=self._vs_configuration)
+
+    def _build_autotools_impl(self, configure_args):
+        win_bash = False
```

Comment:
`win_bash = self._is_mingw` ?

Reply:
I guess, `tools.os_info.is_windows` is better condition these times, there are other cases it's needed (Android on Windows, Clang, etc.)

Reply:
I guess we can add a build requirement on `msys2` when `os_info.is_windows` is `True`, since `autoreconf` and bash scripts are used (for Emscripten, MinGW, etc)

Reply:
yes, lets add